pkg/registry/batch/job/strategy_test.go

```diff
@@ -285,6 +400,28 @@ func TestJobStrategy_PrepareForCreate(t *testing.T) {
 				},
 			},
 		},
+		"job does not allow setting status on create": {
+			enableJobPodFailurePolicy: false,
+			job: batch.Job{
+				ObjectMeta: getValidObjectMeta(0),
+				Spec: batch.JobSpec{
+					Selector:         validSelector,
+					Template:         validPodTemplateSpec,
+					PodFailurePolicy: podFailurePolicy,
```

Comment:
nit: remove this field and the flag `enableJobPodFailurePolicy`. Handling of the field, when the feature is disabled is tested in the test case above.